libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -573,6 +587,14 @@ bool AP_Arming::arm_checks(uint8_t method)
         }
     }
 
+    // check system health on arm as well as prearm
+    if ((checks_to_perform & ARMING_CHECK_ALL) ||
```

Comment:
This is a redundant guard, system_checks already does the check against `checks_to_preform`

Reply:
that's what I thought, but it doesn't work. The pre-arm checks are cached in copter. So if the microSD fails before arming (or is removed), it allows arming with bad storage.

Reply:
ideally we'd have a mask of checks that should be cached, and which ones should be specifically checked again

Reply:
I thought I removed the pre-arm-check-caching. 
7b3e111f85d936db047d81ba5a49d1a22fb0667a

Reply:
it didn't seem to work when I tested today. If I pulled the microSD out a few seconds before arming it allowed me to arm, and ChibiOS::Storage::healthy() wasn't called